src/ApiClient.php

```diff
@@ -39,17 +40,32 @@ class ApiClient
      */
     private $userAgent;
 
+    private CookieJar $cookieJar;
+    private $cookiesCacheFile = __DIR__.'/cookies.txt';
+    private $crumbCacheFile = __DIR__.'/crumb.txt';
```

Comment:
You cannot assume that these files can be written in any environment.

If you want to offer caching for cookies and crumb values, you need to provide configurable caching to users. "Configurable" also means, you cannot assume that people want their cache to be written to a file, so you have to provide some kind of interface. Relying on https://www.php-fig.org/psr/psr-6/ would be a solid choice for caching.